feeds/pypi/pypi.go

```diff
@@ -73,28 +87,97 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
+func fetchCriticalPackages(packageList []string) ([]*Package, error) {
+	responseChannel := make(chan *Response)
+	errs := make(chan error)
+	var wg sync.WaitGroup
+
+	for _, pkgName := range packageList {
+		wg.Add(1)
+		go func(pkgName string) {
+			resp, err := httpClient.Get(fmt.Sprintf(packageURLFormat, pkgName))
+			if err != nil {
+				errs <- err
+				return
```

Comment:
Does this need an `defer wg.Done()` at the start of the func (or explicitly on each error return) to notify the later `wg.Wait()` to stop blocking on `return`? I'm not sure if this behaviour is triggered by the return statement already.